nixos/modules/virtualisation/amazon-image.nix

```diff
@@ -33,23 +33,28 @@ in
 
     boot.growPartition = cfg.hvm;
 
-    fileSystems."/" = {
+    fileSystems."/" = mkIf (!cfg.zfs.enable) {
       device = "/dev/disk/by-label/nixos";
       fsType = "ext4";
       autoResize = true;
     };
 
-    fileSystems."/boot" = mkIf cfg.efi {
+    fileSystems."/boot" = if cfg.efi then {
       device = "/dev/disk/by-label/ESP";
       fsType = "vfat";
-    };
+    } else if cfg.zfs.enable then {
+      device = "boot/boot";
+      fsType = "zfs";
+    } else {};
```

Comment:
This should probably be something like:

```nix
fileSystems."/boot" = mkMerge [
  (mkIf cfg.efi {
    # ...
  })
  (mkIf cfg.zfs.enable {
    # ...
  })
];
```

and then add an assertion for using `cfg.efi` and `cfg.zfs.enable` at the same time.

Reply:
Great catch!

Reply:
Can't ZFS be used with EFI?

Reply:
I'm remembering what is up here. Since it is SBBR it needs to be EFI, but EFI doesn't support ZFS as a boot target, meaning the boot disk needs to be fat32. I didn't use ZFS for the boot disk for ??? reasons initially. I'll replace the boot  disk which is currently zfs with a boot disk which is fat.